File: Behavioral/Mediator/mediator.cs
// Mediator Pattern in C#

using System;
using System.Collections.Generic;

public abstract class Mediator
{
    public abstract void Send(string message, Colleague colleague);
}

public abstract class Colleague
{
    protected Mediator mediator;

    public Colleague(Mediator mediator)
    {
        this.mediator = mediator;
    }
}

public class User : Colleague
{
    private string name;

    public User(string name, Mediator mediator) : base(mediator)
    {
        this.name = name;
    }

    public void Send(string message)
    {
        Console.WriteLine($"{name} sends: {message}");
        mediator.Send(message, this);
    }

    public void Receive(string message)
    {
        Console.WriteLine($"{name} receives: {message}");
    }
}

public class ChatRoom : Mediator
{
    private List<User> users = new List<User>();

    public void AddUser(User user)
    {
        users.Add(user);
    }

    public override void Send(string message, Colleague colleague)
    {
        foreach (User user in users)
        {
            if (user != colleague)
            {
                user.Receive(message);
            }
        }
    }
}

// Usage
class Program
{
    static void Main(string[] args)
    {
        ChatRoom chatRoom = new ChatRoom();
        User user1 = new User("Alice", chatRoom);
        User user2 = new User("Bob", chatRoom);
        chatRoom.AddUser(user1);
        chatRoom.AddUser(user2);

        user1.Send("Hello, Bob!");  // Alice sends: Hello, Bob!
                                    // Bob receives: Hello, Bob!
    }
}

File: Behavioral/Mediator/mediator.py
# Mediator Pattern in Python

class Mediator:
    def send(self, message, colleague):
        pass

class Colleague:
    def __init__(self, mediator):
        self.mediator = mediator

class User(Colleague):
    def __init__(self, name, mediator):
        super().__init__(mediator)
        self.name = name

    def send(self, message):
        print(f"{self.name} sends: {message}")
        self.mediator.send(message, self)

    def receive(self, message):
        print(f"{self.name} receives: {message}")

class ChatRoom(Mediator):
    def __init__(self):
        self.users = []

    def add_user(self, user):
        self.users.append(user)

    def send(self, message, user):
        for u in self.users:
            if u != user:
                u.receive(message)

# Usage
chat_room = ChatRoom()
user1 = User("Alice", chat_room)
user2 = User("Bob", chat_room)
chat_room.add_user(user1)
chat_room.add_user(user2)

user1.send("Hello, Bob!")  # Alice sends: Hello, Bob!
                          # Bob receives: Hello, Bob!